subsys/net/ip/net_core.c

```diff
@@ -381,6 +428,12 @@ static int net_init(struct device *unused)
 {
 	int status = 0;
 
+#if defined(CONFIG_NET_HOSTNAME_ENABLE)
+	memcpy(_hostname, CONFIG_NET_HOSTNAME, sizeof(CONFIG_NET_HOSTNAME) - 1);
+
+	NET_DBG("Hostname set to %s", _hostname);
+#endif
```

Comment:
so here instead it would call net_hostname_init(), a function exposed in net_private.h, something like that.
It will avoid having the ifdef here at least (it will be in net_private.h then).